File: onlinepayments-sdk/src/main/java/com/onlinepayments/sdk/client/android/model/paymentproduct/BasicPaymentItem.java
/*
 * Copyright 2020 Global Collect Services B.V
 */

package com.onlinepayments.sdk.client.android.model.paymentproduct;

import com.onlinepayments.sdk.client.android.model.paymentproduct.displayhints.DisplayHintsPaymentItem;

import java.io.Serializable;
import java.util.List;

public interface BasicPaymentItem extends Serializable {

    String getId();

    /**
     * @deprecated use {@link #getDisplayHintsList()} )}
     */
    @Deprecated
    DisplayHintsPaymentItem getDisplayHints();

    List<DisplayHintsPaymentItem> getDisplayHintsList();

    List<AccountOnFile> getAccountsOnFile();

}

File: sync/Worldline/repo/onlinepayments-sdk/src/main/kotlin/com/onlinepayments/sdk/client/android/model/paymentproduct/BasicPaymentItem.kt
/*
 * Do not remove or alter the notices in this preamble.
 *
 * Copyright © 2025 Worldline and/or its affiliates.
 *
 * All rights reserved. License grant and user rights and obligations according to the applicable license agreement.
 *
 * Please contact Worldline for questions regarding license and user rights.
 */
package com.onlinepayments.sdk.client.android.model.paymentproduct

import com.onlinepayments.sdk.client.android.model.paymentproduct.displayhints.DisplayHintsPaymentItem
import java.io.Serializable

interface BasicPaymentItem : Serializable {
    fun getId(): String?

    fun getDisplayHintsList(): MutableList<DisplayHintsPaymentItem>

    fun getAccountsOnFile(): MutableList<AccountOnFile>
}
